DuplicateAssemblyScanner/DuplicateAssemblyScanner/Util/MemberValue.cs

```diff
@@ -0,0 +1,95 @@
+namespace DuplicateAssemblyScanner.Util {
+    using System;
+    using System.Linq;
+    using System.Reflection;
+
+    /// <summary>
+    /// A nicer way to retrieve values of class instance members.
+    /// Derived from: https://stackoverflow.com/questions/60236067/
+    /// </summary>
+    public class MemberValue {
+
+        /// <summary>
+        /// Default <see cref="BindingFlags"/> to use when retrieving for members.
+        /// </summary>
+        private const BindingFlags DefaultBindingFlags =
+            BindingFlags.Public | BindingFlags.Instance | BindingFlags.Static | BindingFlags.IgnoreCase;
+
+        /// <summary>
+        /// Retrieves the value from a field/method/property member of an object.
+        /// </summary>
+        /// 
+        /// <typeparam name="T">The value <see cref="Type"/>.</typeparam>
+        /// 
+        /// <param name="instance">An <see cref="object"/> instance.</param>
+        /// <param name="member">The <see cref="MemberInfo"/> for the member to inspect.</param>
+        /// 
+        /// <returns>Returns the value of the <paramref name="member"/>, if found, otherwise the <c>default</c> value for <typeparamref name="T"/>.</returns>
+        public static T GetValue<T>(object instance, MemberInfo member) {
+            return member.MemberType switch
+            {
+                MemberTypes.Field => (T)(member as FieldInfo)?.GetValue(instance),
```

Comment:
The null check is not necessary since `MemberType.Field` implies that `member` is `FieldInfo`.

Reply:
Is this also the case for the `Method` and `Property`?

Reply:
I'd completely rewrite the `switch` expression by using the pattern matching:
```C#
return member switch
{
    FieldInfo field => (T)field.GetValue(instance),
    MethodInfo method => (T)method.Invoke(instance, null),
    PropertyInfo property => (T)property.GetValue(instance, null),
    _ => default,
};
```

Reply:
Class is no longer used; removed.